e2e/epinio.e2e.spec.ts

```diff
@@ -0,0 +1,239 @@
+
+export async function installEpinioCli() {
+  const platform = os.platform() as string;
+
+  if (!platforms[platform]) {
+    console.error(`Platform type not detect. Found: ${ platform }`);
+  }
+
+  // Download epinio binary based on platform type
+  await downloadEpinioBinary(platform);
+}
+
+/**
+ * Download epinio binary based on the platform type and save the binary
+ * into a temporary folder.
+ */
+export async function downloadEpinioBinary( platformType: string) {
+  // Setting up epinio binaries names per platform
+  const epinioWin = 'epinio-windows-amd64.exe';
+  const epinioLinux = 'epinio-linux-x86_64';
+  const epinioDarwin = 'epinio-darwin-x86_64';
+  const epinioDarwinArm = 'epinio-darwin-arm64';
+  const epinioWorkingVersion = 'v0.5.0';
+
+  // Create a temp folder for epinio binary
+  const epinioTempFolder = path.join(os.homedir(), 'epinio-tmp');
+
+  if (!fs.existsSync(epinioTempFolder)) {
+    fs.mkdirSync(epinioTempFolder, { recursive: true });
+  }
+
+  // Detect CPU arch
+  const cpuArch = os.arch();
+
+  switch (platformType) {
+  case 'darwin':
+    if (cpuArch === 'x64') {
+      await downloadEpinioCommand(epinioWorkingVersion, epinioDarwin, epinioTempFolder);
+      break;
+    } else {
+      await downloadEpinioCommand(epinioWorkingVersion, epinioDarwinArm, epinioTempFolder);
+      break;
+    }
+  case 'linux':
+    await downloadEpinioCommand(epinioWorkingVersion, epinioLinux, epinioTempFolder);
+    break;
+  case 'win32':
+    await downloadEpinioCommand(epinioWorkingVersion, epinioWin, epinioTempFolder);
+    break;
+  }
+}
+
+/**
+ * Download latest epinio cli binary and makes it executable
+ */
+export async function downloadEpinioCommand(version: string, platform: string, folder: string) {
+  const epinioUrl = 'https://github.com/epinio/epinio/releases/download/';
+
+  if (!os.platform().startsWith('win')) {
+    await curl('--fail', '--location', `${ epinioUrl }${ version }/${ platform }`, '--output', `${ folder }\/epinio`);
+    const stat = fs.statSync(`${ folder }\/epinio`).mode;
+
+    fs.chmodSync(`${ folder }\/epinio`, stat | 0o755);
+  } else {
+    const winPath = path.resolve(folder);
+    await curl('--fail', '--location', `${ epinioUrl }${ version }/${ platform }`, '--output', `${ winPath }\\epinio.zip`);
+    await unzip('-o', `${ winPath }\\epinio.zip`, 'epinio.exe', '-d', `${ folder }`);
+  }
+}
+
+/**
+ * Gracefully remove epinio temp folder and uninstall all epinio-install resources
+ */
+export async function tearDownEpinio() {
+  const epinioTempFolder = path.join(os.homedir(), 'epinio-tmp');
+
+  if (fs.existsSync(epinioTempFolder)) {
+    fs.rmSync(epinioTempFolder, { recursive: true, maxRetries: 10 });
+  }
+
+  await helm('uninstall', 'epinio-installer', '--timeout=20m');
+}
+
+/**
+ * Run the given tool with the given arguments, returning its standard output.
+ */
+export async function tool(tool: string, ...args: string[]): Promise<string> {
```

Comment:
Why are we not using the `TestUtils.ts` version?

Reply:
Don't know, but it seems to work.